### PR TITLE
chore(j-s): Limit Prison System Data

### PR DESCRIPTION
**Pull request by gudjong**
# Limit Prison System Data

[Takmarka hvaða svæði eru aðgengileg fangelsum og FMST í bakenda](https://app.asana.com/0/1199153462262248/1204522261372979/f)

## What

- Limits the data delivered to the client for prison system users. This is similar to data delivery to defence users.

## Why

- Improved security.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/components/IndictmentCaseFilesList/IndictmentCaseFilesList.tsx [hunk 1]
```diff
@@ -137,7 +137,7 @@ const IndictmentCaseFilesList: React.FC<React.PropsWithChildren<Props>> = (
               <PdfButton
                 caseId={workingCase.id}
                 title={formatMessage(caseFiles.trafficViolationIndictmentTitle)}
-                pdfType={`${limitedAccess ? 'limitedAccess/' : ''}indictment`}
+                pdfType={'indictment'}
```
Comment (unakb) [inline apps/judicial-system/web/src/components/IndictmentCaseFilesList/IndictmentCaseFilesList.tsx:140]: Nice, much cleaner

**Review by unakb — APPROVED** [review]
Can't see any glaring errors, but we should regression test this well for sure

File: apps/judicial-system/backend/src/app/modules/case/filters/case.filter.ts [hunk 2]
```diff
@@ -75,16 +69,10 @@ function canProsecutionUserAccessCase(
...
 function canDistrictCourtUserAccessCase(theCase: Case, user: User): boolean {
   // Check case type access
-  if ([UserRole.JUDGE, UserRole.REGISTRAR].includes(user.role)) {
-    if (
-      !isRestrictionCase(theCase.type) &&
-      !isInvestigationCase(theCase.type) &&
-      !isIndictmentCase(theCase.type)
-    ) {
+  if (![UserRole.JUDGE, UserRole.REGISTRAR].includes(user.role)) {
+    if (!indictmentCases.includes(theCase.type)) {
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/filters/case.filter.ts:73]: `isIndictmentCases` is better and will change it back.

And yes, assistants are only allowed to see indictment cases. R-cases are more sensitive.

File: apps/judicial-system/backend/src/app/modules/case/filters/case.filter.ts [hunk 3]
```diff
@@ -177,15 +165,91 @@ function canPrisonSystemUserAccessCase(
...
+      )
+    ) {
+      return false
+    }
+  } else {
+    if (theCase.defenderNationalId !== user.nationalId) {
+      return false
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/filters/case.filter.ts:252]: We do that in the guards, but in the case filter I find it very difficult to do. Maybe we can take a look at this during a cooldown week in the future.

**Pull request merged**